test/system/src/test/java/io/pravega/test/system/ReadWriteAndScaleWithFailoverTest.java

```diff
@@ -0,0 +1,195 @@
+        Service controllerService = new PravegaControllerService("controller", zkUri);
+        if (!controllerService.isRunning()) {
+            controllerService.start(true);
+        }
+        controllerService.scaleService(3, true);
+        List<URI> conUris = controllerService.getServiceDetails();
+        log.debug("Pravega Controller service  details: {}", conUris);
+
+        // Fetch all the RPC endpoints and construct the client URIs.
+        final List<String> uris = conUris.stream().filter(uri -> uri.getPort() == 9092).map(URI::getAuthority)
+                .collect(Collectors.toList());
+
+        URI controllerURI = URI.create("tcp://" + String.join(",", uris));
+        log.info("Controller Service direct URI: {}", controllerURI);
+
+        //4.start 3 instances of pravega segmentstore
+        Service segService = new PravegaSegmentStoreService("segmentstore", zkUri, controllerURI);
+        if (!segService.isRunning()) {
+            segService.start(true);
+        }
+        segService.scaleService(3, true);
+        List<URI> segUris = segService.getServiceDetails();
+        log.debug("Pravega Segmentstore service  details: {}", segUris);
+    }
+
+    @Before
+    public void setup() {
+        // Get zk details to verify if controller, SSS are running
+        Service zkService = new ZookeeperService("zookeeper");
+        List<URI> zkUris = zkService.getServiceDetails();
+        log.debug("Zookeeper service details: {}", zkUris);
+        //get the zk ip details and pass it to  host, controller
+        URI zkUri = zkUris.get(0);
+
+        // Verify controller is running.
+        controllerInstance = new PravegaControllerService("controller", zkUri);
+        assertTrue(controllerInstance.isRunning());
+        List<URI> conURIs = controllerInstance.getServiceDetails();
+        log.info("Pravega Controller service instance details: {}", conURIs);
+
+        // Fetch all the RPC endpoints and construct the client URIs.
+        final List<String> uris = conURIs.stream().filter(uri -> uri.getPort() == 9092).map(URI::getAuthority)
+                .collect(Collectors.toList());
+
+        controllerURIDirect = URI.create("tcp://" + String.join(",", uris));
+        log.info("Controller Service direct URI: {}", controllerURIDirect);
+
+        // Verify segment store is running.
+        segmentStoreInstance = new PravegaSegmentStoreService("segmentstore", zkUri, controllerURIDirect);
+        assertTrue(segmentStoreInstance.isRunning());
+        log.info("Pravega Segmentstore service instance details: {}", segmentStoreInstance.getServiceDetails());
+
+        //executor service
+        executorService = Executors.newFixedThreadPool(NUM_READERS + NUM_WRITERS);
+        //get Controller Uri
+        controller = new ControllerImpl(controllerURIDirect);
+        testState = new TestState();
+    }
+
+    @After
+    public void tearDown() {
+        controllerInstance.scaleService(1, true);
```

Comment:
we are reducing the controller, segmentstore instances to 1, so that other system tests can use the same.